### PR TITLE
[bug] break when empty Timeline

### PR DESCRIPTION
**Pull request by kimdwkimdw**
I figured out before `__post_init__`, checking emptiness is useful.

### CODE REVIEW

**Comment by kimdwkimdw** [comment]
@hbredin 
This PR can be helpful.
https://github.com/pyannote/pyannote-core/issues/69

File: pyannote/core/segment.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def __bool__(self):
 
     def __post_init__(self):
         """Round start and end up to SEGMENT_PRECISION precision (when required)"""
-        if AUTO_ROUND_TIME:
+        if self and AUTO_ROUND_TIME:
```
Comment (hbredin) [inline pyannote/core/segment.py:173]: As for the original `Segment.set_precision` PR, I am concerned about speed.
This should avoid unnecessary calls to `__bool__` when `AUTO_ROUND_TIME` is already `False`: 
```suggestion
        if AUTO_ROUND_TIME and self:
```

Can you please double check that it is actually faster?

Comment (kimdwkimdw) [reply pyannote/core/segment.py:173]: great idea

**Comment by hbredin** [comment]
Latest commit should fix this. Can you please try with up-to-date develop branch and confirm?

**Comment by kimdwkimdw** [comment]
Checked! Thanks

How about preserving test-case only?
I tested #71  with this test-case

```
def test_empty_gaps():
    empty_timeline = Timeline(uri='MyAudioFile')
    assert list(empty_timeline.gaps()) == []
    Segment.set_precision(3)
    assert list(empty_timeline.gaps()) == []
```

**Comment by hbredin** [comment]
I thought about this before pushing my fix. 

I am just a bit worried of the side effect of using `Segment.set_precision` on other subsequent tests.
We could call `Segment.set_precision(None)` at the end of`test_empty_gaps` but what if it fails before reaching this line? Any idea?

**Comment by kimdwkimdw** [comment]
> I thought about this before pushing my fix.
> 
> I am just a bit worried of the side effect of using `Segment.set_precision` on other subsequent tests. We could call `Segment.set_precision(None)` at the end of`test_empty_gaps` but what if it fails before reaching this line? Any idea?

https://github.com/pyannote/pyannote-core/pull/72

I made another proposal.

**Pull request closed without merging**